tests/ui/integer_arithmetic.stderr

```diff
@@ -12,158 +12,182 @@ error: this operation will panic at runtime
 LL |     i %= 0;
    |     ^^^^^^ attempt to calculate the remainder of `_` with a divisor of zero
 
-error: integer arithmetic detected
+error: arithmetic detected
   --> $DIR/integer_arithmetic.rs:9:5
    |
 LL |     1 + i;
    |     ^^^^^
    |
-   = note: `-D clippy::integer-arithmetic` implied by `-D warnings`
+   = note: `-D clippy::arithmetic` implied by `-D warnings`
 
-error: integer arithmetic detected
+error: arithmetic detected
   --> $DIR/integer_arithmetic.rs:10:5
    |
 LL |     i * 2;
    |     ^^^^^
 
-error: integer arithmetic detected
+error: arithmetic detected
   --> $DIR/integer_arithmetic.rs:11:5
    |
 LL | /     1 %
 LL | |     i / 2; // no error, this is part of the expression in the preceding line
-   | |_____^
+   | |_________^
 
-error: integer arithmetic detected
+error: arithmetic detected
   --> $DIR/integer_arithmetic.rs:13:5
    |
 LL |     i - 2 + 2 - i;
    |     ^^^^^^^^^^^^^
 
-error: integer arithmetic detected
+error: arithmetic detected
   --> $DIR/integer_arithmetic.rs:14:5
    |
 LL |     -i;
    |     ^^
 
-error: integer arithmetic detected
+error: arithmetic detected
   --> $DIR/integer_arithmetic.rs:15:5
    |
 LL |     i >> 1;
    |     ^^^^^^
 
-error: integer arithmetic detected
+error: arithmetic detected
   --> $DIR/integer_arithmetic.rs:16:5
    |
 LL |     i << 1;
    |     ^^^^^^
 
-error: integer arithmetic detected
+error: arithmetic detected
   --> $DIR/integer_arithmetic.rs:26:5
    |
 LL |     i += 1;
    |     ^^^^^^
 
-error: integer arithmetic detected
+error: arithmetic detected
   --> $DIR/integer_arithmetic.rs:27:5
    |
 LL |     i -= 1;
    |     ^^^^^^
 
-error: integer arithmetic detected
+error: arithmetic detected
   --> $DIR/integer_arithmetic.rs:28:5
    |
 LL |     i *= 2;
    |     ^^^^^^
 
-error: integer arithmetic detected
-  --> $DIR/integer_arithmetic.rs:31:11
+error: arithmetic detected
+  --> $DIR/integer_arithmetic.rs:29:5
+   |
+LL |     i /= 2;
+   |     ^^^^^^
+
+error: arithmetic detected
+  --> $DIR/integer_arithmetic.rs:30:5
+   |
+LL |     i /= 0;
+   |     ^^^^^^
+
+error: arithmetic detected
+  --> $DIR/integer_arithmetic.rs:31:5
```

Comment:
Replacing `#![warn(clippy::integer_arithmetic, clippy::float_arithmetic)]` with `#![warn(clippy::arithmetic)]` actually improved diagnostics.

If you guys want to deprecate `integer_arithmetic` and `float_arithmetic` in favour of `arithmetic`, then I can scan for primitives to recover the lost "integer/float arithmetic detected" messages. Although that would imply more processing and such thing doesn't seem too useful at all.

Reply:
`integer_arithmetic` specifically checks for uses which may panic. `float_arithmetic` could maybe be done, but that can be a later problem.

Reply:
`i /= 0` and `i %= 0` will panic when the compiler is not able to deduct the RHS at compile-time so why the lint currently omits them?

Regardless, I reverted these changes for now but they can be reviewed in the future.

Reply:
Both `/= 0` and `%= 0` are caught by the `unconditional_panic` lint.